.github/workflows/run_tests_osx.yml

```diff
@@ -478,3 +478,51 @@ jobs:
           cd build
           LD_LIBRARY_PATH=${LD_LIBRARY_PATH} ctest -j 12 --rerun-failed --output-on-failure -VV
         if: ${{ failure() }}
+
+  fetch_content_cmake:
+    needs: build-deps-osx
+    runs-on: macos-12
+    strategy:
+      matrix:
+        hdf5: [ 1.12.2, 1.14.3 ]
+    steps:
+      - uses: actions/checkout@v3
+
+    ###
+    # Set Environmental Variables
+    ###
+
+      - run: echo "CMAKE_PREFIX_PATH=${HOME}/environments/${{ matrix.hdf5 }}/" >> $GITHUB_ENV
+      - run: echo "LD_LIBRARY_PATH=${HOME}/environments/${{ matrix.hdf5 }}/lib" >> $GITHUB_ENV
```

Comment:
Not a fan of these approaches, but I understand the need for them right now. If hdf5 is fetch-contentable, this could be simplified greatly.

Anyway, `LD_LIBRARY_PATH` should be unnecessary if running from build directory and the dependencies are installed with RPATH. Also, for mac it's `DYLD_LIBRARY_PATH`